paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/DataTableSource.java

```diff
@@ -198,13 +202,45 @@ public ScanRuntimeProvider getScanRuntimeProvider(ScanContext scanContext) {
                         .withProjection(projectFields)
                         .withPredicate(predicate)
                         .withLimit(limit)
-                        .withParallelism(
-                                Options.fromMap(table.schema().options())
-                                        .get(FlinkConnectorOptions.SCAN_PARALLELISM))
                         .withWatermarkStrategy(watermarkStrategy);
 
         return new PaimonDataStreamScanProvider(
-                !streaming, env -> sourceBuilder.withEnv(env).build());
+                !streaming,
+                env -> {
+                    Tuple2<Integer, List<Split>> scanContent = scanContent();
+                    return sourceBuilder
+                            .withParallelism(scanContent.f0)
+                            .withSplits(scanContent.f1)
+                            .withEnv(env)
+                            .build();
+                });
+    }
+
+    private Tuple2<Integer, List<Split>> scanContent() {
```

Comment:
maybe `void configureSource(SourceBuilder source)` here?

Reply:
I update it and fix the conflict